modules/ModuleBuilder/parsers/views/AbstractMetaDataImplementation.php

```diff
@@ -353,12 +353,27 @@ protected function _saveToFile($filename, $defs, $useVariables = true, $forPopup
 
         $out .= ";\n?>\n";
 
-
         if (sugar_file_put_contents($filename, $out) === false) {
             $GLOBALS ['log']->fatal(get_class($this) . ": could not write new viewdef file " . $filename);
         }
     }
 
+    /**
+     * @param $defs The definitions to save
+     * @return bool
+     */
+    private function shouldAppendOriginalViewTemplateDefs($defs)
+    {
+        if (empty($this->_originalViewTemplateDefs)) {
+            return false;
+        }
+        if (is_array($defs) && isset($defs[$this->_viewName])) {
+            // The defs are already being saved we don't want to duplicate them
+            return false;
+        }
```

Comment:
You are able to merge these conditions.

Reply:
True, but I think it's more readable like this.

Reply:
:D